central/graphql/resolvers/node_vulnerabilities_test.go

```diff
@@ -0,0 +1,600 @@
+	impact, err := vuln.EnvImpact(ctx)
+	s.NoError(err)
+	s.Equal(1.0, impact)
+
+	vuln = s.getNodeVulnerabilityResolver(ctx, "cve-2017-1#")
+	impact, err = vuln.EnvImpact(ctx)
+	s.NoError(err)
+	s.Equal(0.5, impact)
+}
+
+func (s *GraphQLNodeVulnerabilityTestSuite) TestNodeVulnerabilityLastScanned() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	vuln := s.getNodeVulnerabilityResolver(ctx, "cve-2018-1#")
+	node := s.getNodeResolver(ctx, "id2")
+	lastScanned, err := vuln.LastScanned(ctx)
+	s.NoError(err)
+	expected, err := timestamp(node.data.GetScan().GetScanTime())
+	s.NoError(err)
+	s.Equal(expected, lastScanned)
+}
+
+func (s *GraphQLNodeVulnerabilityTestSuite) TestNodeVulnerabilityNodes() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	vuln := s.getNodeVulnerabilityResolver(ctx, "cve-2018-1#")
+
+	nodes, err := vuln.Nodes(ctx, PaginatedQuery{})
+	s.NoError(err)
+	s.Equal(2, len(nodes))
+	idList := getIDList(ctx, nodes)
+	s.ElementsMatch(idList, []string{"id1", "id2"})
+
+	count, err := vuln.NodeCount(ctx, RawQuery{})
+	s.NoError(err)
+	s.Equal(int32(len(nodes)), count)
+
+	vuln = s.getNodeVulnerabilityResolver(ctx, "cve-2017-1#")
+
+	nodes, err = vuln.Nodes(ctx, PaginatedQuery{})
+	s.NoError(err)
+	s.Equal(1, len(nodes))
+	idList = getIDList(ctx, nodes)
+	s.ElementsMatch(idList, []string{"id2"})
+
+	count, err = vuln.NodeCount(ctx, RawQuery{})
+	s.NoError(err)
+	s.Equal(int32(len(nodes)), count)
+}
+
+func (s *GraphQLNodeVulnerabilityTestSuite) TestNodeVulnerabilityNodeComponents() {
+	ctx := SetAuthorizerOverride(s.ctx, allow.Anonymous())
+
+	vuln := s.getNodeVulnerabilityResolver(ctx, "cve-2018-1#")
+
+	comps, err := vuln.NodeComponents(ctx, PaginatedQuery{})
+	s.NoError(err)
+	s.Equal(2, len(comps))
+	idList := getIDList(ctx, comps)
+	s.ElementsMatch(idList, []string{"comp1#0.9#", "comp2#1.1#"})
+
+	count, err := vuln.NodeComponentCount(ctx, RawQuery{})
+	s.NoError(err)
+	s.Equal(int32(len(comps)), count)
+
+	vuln = s.getNodeVulnerabilityResolver(ctx, "cve-2017-1#")
+
+	comps, err = vuln.NodeComponents(ctx, PaginatedQuery{})
+	s.NoError(err)
+	s.Equal(1, len(comps))
+	idList = getIDList(ctx, comps)
+	s.ElementsMatch(idList, []string{"comp4#1.0#"})
+
+	count, err = vuln.NodeComponentCount(ctx, RawQuery{})
+	s.NoError(err)
+	s.Equal(int32(len(comps)), count)
+}
+
+func (s *GraphQLNodeVulnerabilityTestSuite) getNodeResolver(ctx context.Context, id string) *nodeResolver {
+	nodeID := graphql.ID(id)
+
+	node, err := s.resolver.Node(ctx, struct{ graphql.ID }{nodeID})
+	s.NoError(err)
+	s.Equal(nodeID, node.Id(ctx))
+	return node
+}
+
+func (s *GraphQLNodeVulnerabilityTestSuite) getNodeVulnerabilityResolver(ctx context.Context, id string) NodeVulnerabilityResolver {
+	vulnID := graphql.ID(id)
+
+	vuln, err := s.resolver.NodeVulnerability(ctx, IDQuery{ID: &vulnID})
+	s.NoError(err)
+	s.Equal(vulnID, vuln.Id(ctx))
+	return vuln
+}
+
+func testNodes() []*storage.Node {
```

Comment:
Could you please move this function to the `test_utils.go` file, this could be reused in future tests on the node endpoint

Reply:
Yeah that's right. Moved the function.